### PR TITLE
New package: pal-0.4.3

### PR DESCRIPTION
**Pull request by dkociela**

### CODE REVIEW

File: srcpkgs/pal/template [hunk 1]
```diff
@@ -0,0 +1,20 @@
+pkgname="pal"
+version="0.4.3"
+revision=1
+build_wrksrc=src
+only_for_archs="x86_64 i686"
```
Comment (xtraeme) [inline srcpkgs/pal/template]: Stop using this to make travis report a success, this is not the solution.

File: srcpkgs/pal/template [hunk 2]
```diff
@@ -0,0 +1,20 @@
+pkgname="pal"
```
Comment (xtraeme) [inline srcpkgs/pal/template]: The first line should be a comment.

File: srcpkgs/pal/template [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+version="0.4.3"
+revision=1
+build_wrksrc=src
+only_for_archs="x86_64 i686"
+hostmakedepends="pkg-config"
+makedepends="glib-devel gettext-devel readline-devel"
```
Comment (xtraeme) [inline srcpkgs/pal/template]: - `gettext-devel` should be in `hostmakedepends` instead.

File: srcpkgs/pal/template [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+homepage="http://palcal.sourceforge.net/"
+distfiles="${SOURCEFORGE_SITE}/project/palcal/$pkgname/$version/$pkgname-$version.tgz"
+checksum="ce470cb7be76522ff58cd0325ad7817b6cc2132a5e0cea55de8c9eb63b54551c"
+
+do_build() {
+	make
+}
+do_install() {
+	make DESTDIR=${DESTDIR} install
+}
```
Comment (xtraeme) [inline srcpkgs/pal/template]: Use `build_style=gnu-makefile` instead.

File: srcpkgs/pal/template (hunk 1 above)
Comment (dkociela) [inline srcpkgs/pal/template]: OK, thanks. Just curious though, what is only_for_archs for?

File: srcpkgs/pal/template (hunk 3 above)
Comment (dkociela) [inline srcpkgs/pal/template]: Thanks for the clue. I had to move a couple of things around between
hostmakedepends
and makedepends but finally got it.

File: srcpkgs/pal/template (hunk 4 above)
Comment (dkociela) [inline srcpkgs/pal/template]: Thanks, I did try that but got build failures that I couldn't work around.

**Comment by dkociela** [comment]
I finally had to set "noarch" to get success. I wonder if I should have set "nostrip" instead.

**Comment by pullmoll** [comment]
Setting `noarch=yes` for a package which compiles binaries is certainly wrong.
If you can't manage to get cross compiles working, you should just set `nocross=yes`.
If compiling for musl libc fails, just leave it as is. Perhaps someone else knows how to fix it and does it later on. Or perhaps it just won't work with musl libc.

**Comment by dkociela** [comment]
Everything goes well until the post-install phase (armv6, v7), running the strip/debug hook. Strip exits with "/usr/bin/pal: file format not recognized".
Setting "nostrip" in the template, everything succeeds. 
The package builds and installs fine on my x86-64 sytem, and the binary "/usr/bin/pal" shows as "ELF 64-bit LSB executable, x86-64"
I'm still trying to figure out why "strip' gives that error (only on the arm arch)

**Comment by dkociela** [comment]
Thanks, your comment clarified something for me.

File: srcpkgs/pal/template [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+build_wrksrc=src
+hostmakedepends="pkg-config gettext-devel ncurses-devel readline-devel glib-devel"
+makedepends="glib-devel"
+short_desc="Command line calendar that can keep track of events"
+maintainer="Dan Kociela <dkociela@gmail.com>"
+license="GPL-2"
+homepage="http://palcal.sourceforge.net/"
+distfiles="${SOURCEFORGE_SITE}/project/palcal/$pkgname/$version/$pkgname-$version.tgz"
+checksum="ce470cb7be76522ff58cd0325ad7817b6cc2132a5e0cea55de8c9eb63b54551c"
+do_build() {
```
Comment (xtraeme) [inline srcpkgs/pal/template]: You are just duplicating what the `gnu-makefile` build_style already does, please remove `do_{build,install}` and set `build_style=gnu-makefile`

File: srcpkgs/pal/template [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# Template file for 'pal'
+
```
Comment (Vaelatern) [inline srcpkgs/pal/template]: No newline here

File: srcpkgs/pal/template (hunk 2 above)
Comment (Vaelatern) [inline srcpkgs/pal/template]: This was taken care of

**Comment by Vaelatern** [comment]
Is this pull request abandoned?

**Comment by dkociela** [comment]
I couldn't get it to cross compile for arm without setting `nostrip=yes` . Also the build fails when using 
`build_style=gnu-makefile`.
I guess I've given up. Should I close it?

**Comment by Vaelatern** [comment]
Don't close it, I'll take more of a stab at it. You are right about it being hard to build with the `build_style`.

**Comment by Vaelatern** [comment]
At present, this package requires headers like `glib.h` which are not even present on Void, and even has a smattering of specific types like `gint` and `gchar` and calls functions like `g_free()`. I don't see this as getting in the repos without impressive work.

I recommend using `wyrd` instead.

**Pull request closed without merging**